documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+# Migrate from v1 to v2 of the AWS IoT SDK for JavaScript
+
+The AWS IoT SDK for JavaScript v2 is a major rewrite of the AWS IoT Device SDK for JavaScript v1 code base. It includes many updates, such as improved
+consistency, ease of use, more detailed information about client status. This guide describes the major features that
+are new in the v2 SDK, and provides guidance on how to migrate your code to v2 from v1 of the AWS IoT SDK for JavaScript.
+
+> [!NOTE]
+> If you can't find the information you need in this guide, visit the [How to get help](#how-to-get-help) section for more help and guidance.
+
+* [What's new in AWS IoT Device SDK for JavaScript v2](#whats-new-in-aws-iot-device-sdk-for-javascript-v2)
+* [How to get started with AWS IoT Device SDK for JavaScript v2](#how-to-get-started-with-aws-iot-device-sdk-for-javascript-v2)
+    * [Package name change](#package-name-change)
+    * [MQTT protocol](#mqtt-protocol)
+    * [Browser applications](#browser-applications)
+    * [Client builder](#client-builder)
+    * [Client start](#client-start)
+    * [Connection types and features](#connection-types-and-features)
+    * [Lifecycle events](#lifecycle-events)
+    * [Publish](#publish)
+    * [Subscribe](#subscribe)
+    * [Unsubscribe](#unsubscribe)
+    * [Client stop](#client-stop)
+    * [Client shutdown](#client-shutdown)
+    * [Reconnects](#reconnects)
+    * [Offline operations queue](#offline-operations-queue)
+    * [Logging](#logging)
+    * [Client for AWS IoT Device Shadow](#client-for-aws-iot-device-shadow)
+    * [Client for AWS IoT Jobs](#client-for-aws-iot-jobs)
+    * [Client for AWS IoT fleet provisioning](#client-for-aws-iot-fleet-provisioning)
+    * [Example](#example)
+* [How to get help](#how-to-get-help)
+* [Appendix](#appendix)
+    * [MQTT5 features](#mqtt5-features)
+
+> [!NOTE]
+> There are multiple v2 references in the v1 SDK documentation. These references does **not** refer to the v2 SDK, but to the release versions of the v1 SDK after v2.0.0.
```

Comment:
Trivial: does -> do

Wondering if we should clarify the v1 SDK documentation and its references to v2 rather than add this note here.

Reply:
`V2` is mentioned in V1 SDK issues: e.g. [V2.* doesn't work with node 4.2.6](https://github.com/aws/aws-iot-device-sdk-js/issues/425). But maybe you're right, I'll take another look at it.